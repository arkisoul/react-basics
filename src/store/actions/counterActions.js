export const INCREMENT = "INCREMENT";
export const DECREMENT = "DECREMENT";
export const INCREMENTBYVALUE = "INCREMENTBYVALUE";

// Action creators are functions that returns an action object
export const incrementCounter = () => ({ type: INCREMENT });
export const decrementCounter = () => ({ type: DECREMENT });
export const incrementCounterByValue = (value) => ({
  type: INCREMENTBYVALUE,
  payload: { value },
});
