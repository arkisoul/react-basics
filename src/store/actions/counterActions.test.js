import { describe, it, expect } from "@jest/globals";

import * as counterActions from "./counterActions";

describe("counterActions", () => {
  it("should return increment action object", () => {
    expect(counterActions.incrementCounter()).toEqual({
      type: counterActions.INCREMENT,
    });
    expect(counterActions.incrementCounter()).toMatchSnapshot();
  });

  it("should return decrement action object", () => {
    expect(counterActions.decrementCounter()).toEqual({
      type: counterActions.DECREMENT,
    });
    expect(counterActions.decrementCounter()).toMatchSnapshot();
  });

  it("should return increment action object with payload", () => {
    const value = 5;
    expect(counterActions.incrementCounterByValue(value)).toEqual({
      type: counterActions.INCREMENTBYVALUE,
      payload: { value },
    });
    expect(counterActions.incrementCounterByValue(value)).toMatchSnapshot();
  });
});
