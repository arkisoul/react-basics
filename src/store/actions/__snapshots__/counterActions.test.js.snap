// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`counterActions should return decrement action object 1`] = `
Object {
  "type": "DECREMENT",
}
`;

exports[`counterActions should return increment action object 1`] = `
Object {
  "type": "INCREMENT",
}
`;

exports[`counterActions should return increment action object with payload 1`] = `
Object {
  "payload": Object {
    "value": 5,
  },
  "type": "INCREMENTBYVALUE",
}
`;
